modules/apps/commerce/headless/headless-commerce/headless-commerce-admin-catalog-impl/src/main/java/com/liferay/headless/commerce/admin/catalog/internal/resource/v1_0/SkuResourceImpl.java

```diff
@@ -187,7 +187,23 @@ public Page<Sku> getSkusPage(
 
 	@Override
 	public Sku patchSku(Long id, Sku sku) throws Exception {
-		_updateSKU(_cpInstanceService.getCPInstance(id), sku);
+		CPInstance cpInstance = _cpInstanceService.getCPInstance(id);
+
+		if (cpInstance == null) {
+			throw new NoSuchCPInstanceException(
+				"Unable to find SKU with id " + id);
+		}
+
+		String skuExternalReferenceCode = sku.getExternalReferenceCode();
+
+		if ((skuExternalReferenceCode != null) &&
+			!skuExternalReferenceCode.isEmpty()) {
+
+			cpInstance = _cpInstanceService.updateExternalReferenceCode(
+				skuExternalReferenceCode, cpInstance.getCPInstanceId());
+		}
+
+		_updateSKU(cpInstance, sku);
```

Comment:
Why do you perform an update of the whole instance? Isn't this endpoint suppose to update only the external reference code?